tests/ecc.rs

```diff
@@ -304,3 +304,202 @@ fn mul_point_works() {
             .expect_err("circuit is not satisfied");
     }
 }
+
+#[test]
+fn assert_equal_point_works() {
+    let rng = &mut StdRng::seed_from_u64(8349u64);
+
+    let n = 1 << 4;
+    let label = b"demo";
+    let pp = PublicParameters::setup(n, rng).expect("failed to create pp");
+
+    pub struct DummyCircuit {
+        point: JubJubExtended,
+    }
+
+    impl Default for DummyCircuit {
+        fn default() -> Self {
+            Self {
+                point: dusk_jubjub::GENERATOR_EXTENDED,
+            }
+        }
+    }
+
+    impl Circuit for DummyCircuit {
+        fn circuit<C>(&self, composer: &mut C) -> Result<(), Error>
+        where
+            C: Composer,
+        {
+            // test: generator = generator
+            let w_p1 = composer.append_point(self.point);
+            let w_p2 = composer.append_point(self.point);
+            composer.assert_equal_point(w_p1, w_p2);
+
+            // test: 42 * generator = 42 * generator
```

Comment:
personally, I'd prefer to hardcode one of the points and compute the other one. But that's a nit.